java/runtime/src/main/java/io/ray/runtime/util/ConcurrencyGroupUtils.java

```diff
@@ -0,0 +1,90 @@
+package io.ray.runtime.util;
+
+import com.google.common.base.Preconditions;
+import io.ray.api.Ray;
+import io.ray.api.concurrencygroup.ConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroup;
+import io.ray.api.concurrencygroup.annotations.DefConcurrencyGroups;
+import io.ray.api.concurrencygroup.annotations.UseConcurrencyGroup;
+import io.ray.api.function.RayFuncR;
+import io.ray.runtime.ConcurrencyGroupImpl;
+import io.ray.runtime.RayRuntimeInternal;
+import io.ray.runtime.functionmanager.JavaFunctionDescriptor;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/// TODO: cache this.
+public final class ConcurrencyGroupUtils {
+
+  public static List<ConcurrencyGroup> extractConcurrencyGroupsByAnnotations(
+      RayFuncR<?> actorConstructorLambda) {
+    SerializedLambda serializedLambda = LambdaUtils.getSerializedLambda(actorConstructorLambda);
+    Class<?> actorClz = getReturnTypeFromSignature(serializedLambda.getInstantiatedMethodType());
```

Comment:
What's the difference between `getImplMethodSignature` and `getInstantiatedMethodType`? I see that `getImplMethodSignature` is used in `FunctionManager`.

Reply:
In fact, I'm not very familiar with them. Not quite understand the `type variable` according the document.

```java

    /**
     * Get the signature of the primary functional interface method
     * after type variables are substituted with their instantiation
     * from the capture site.
     * @return the signature of the primary functional interface method
     * after type variable processing
     */
    public final String getInstantiatedMethodType() {
        return instantiatedMethodType;
    }
```
@chaokunyang CC

Reply:
Seems `getInstantiatedMethodType` will capture more generics information